src/containerapp/azext_containerapp/_decorator_utils.py

```diff
@@ -81,6 +81,32 @@ def process_loaded_yaml(yaml_containerapp):
     return yaml_containerapp
 
 
+def process_loaded_yaml_for_connected_env_dapr(yaml_dapr_component):
+    if not isinstance(yaml_dapr_component, dict):  # pylint: disable=unidiomatic-typecheck
+        raise ValidationError('Invalid YAML provided. Please see https://aka.ms/azure-container-apps-yaml for a valid containerapps YAML spec.')
+    if not yaml_dapr_component.get('properties'):
+        yaml_dapr_component['properties'] = {}
+
+    nested_properties = ["componentType",
+                         "version",
+                         "ignoreErrors",
+                         "initTimeout",
+                         "secrets",
+                         "secretStoreComponent",
+                         "metadata",
+                         "scopes",
+                         "serviceComponentBind",
+                         "provisioningState",
+                         "deploymentErrors"]
+    for nested_property in nested_properties:
```

Comment:
properties have to be hard-coded here?

Reply:
Yes, this is consistent with the way we handle `ContainerApp` class.

Reply:
I think we need to think a smarter way to handle it. If there is new property added, it may be missing here

Reply:
We can get the properties from the class ConnectedEnvironmentDaprComponentProperties. I will fix it.